hazelcast/src/main/java/com/hazelcast/config/NearCacheConfig.java

```diff
@@ -92,6 +92,7 @@
 
     private String name = "default";
     private InMemoryFormat inMemoryFormat = DEFAULT_MEMORY_FORMAT;
+    private boolean serializeKeys = true;
```

Comment:
I'm not sure if all code parts will work until the whole serialization implementation is not in place, so for the first step is to keep the old behavior (serialize keys). Then the data structure implementations should follow. After that the `StaleReadDetector` fixes. In the end we can switch the default to `false` and all tests etc. should still work :)